packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1231 @@
+    BoolTypeTemplate: &BoolTypeTemplate !!js/function >
+        (literal) => {
+            return literal.toLowerCase();
+        }
+    IntegerTypeTemplate: &IntegerTypeTemplate null # args: literal, argType (i.e. '_string', '_decimal' etc)
+    DecimalTypeTemplate: &DecimalTypeTemplate null # args: literal, argType
+    LongBasicTypeTemplate: &LongBasicTypeTemplate null # args: literal, argType
+    HexTypeTemplate: &HexTypeTemplate null # args: literal, argType
+    OctalTypeTemplate: &OctalTypeTemplate !!js/function >
+        (literal) => {
+            let offset = 0;
+
+            if (
+                literal.charAt(0) === '0' &&
+                (literal.charAt(1) === '0' || literal.charAt(1) === 'o' || literal.charAt(1) === 'O')
+            ) {
+                offset = 2;
+            } else if (literal.charAt(0) === '0') {
+                offset = 1;
+            }
+
+            literal = `0${literal.substr(offset, literal.length - 1)}`;
+
+            return literal;
+        }
+    NumericTypeTemplate: &NumericTypeTemplate null # args: literal, argType
+    ArrayTypeTemplate: &ArrayTypeTemplate !!js/function >
+        (literal, depth) => {
+            if (literal === '') {
+                return '[]'
+            }
+            return `[${literal}]`;
+        }
+    ArrayTypeArgsTemplate: &ArrayTypeArgsTemplate null # Args: single array element, nestedness, lastElement? (note: not being used atm)
+    NullTypeTemplate: &NullTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    UndefinedTypeTemplate: &UndefinedTypeTemplate !!js/function >
+        () => {
+            return 'null';
+        }
+    ObjectTypeTemplate: &ObjectTypeTemplate !!js/function >
+        (literal) => {
+            return `${literal}`;
+        }
+    ObjectTypeArgsTemplate: &ObjectTypeArgsTemplate !!js/function >
+        (args, depth) => {
+            let isObj = true;
+        
+            if (args.length === 0) {
+                return `[]`;
+            }
+
+            const checkIndex = (strIndex, numericIndex) => {
+                let parsedIndex = parseInt(strIndex);
+                if (isNaN(parsedIndex) || ('' + parsedIndex !== strIndex) || parsedIndex != numericIndex) return false;
+                return true;
+            }
+        
+            let indexTest = -1;
+            let pairs = args.map((arg) => {
+                indexTest++;
+                if (isObj && !checkIndex(arg[0], indexTest)) {
+                    isObj = false;
+                }
+                return `${this.utils.stringifyWithSingleQuotes(arg[0])} => ${arg[1]}`;
+            }).join(', ');
+
+            // Rebuilding pairs for numeric sequential indexes without quotes
+            if (isObj) {
+                pairs = args.map((arg) => {
+                    return `${arg[0]} => ${arg[1]}`;
+                }).join(', ');
+            }
+
+            return `${isObj ? '(object) ' : ''}[${pairs}]`
+        }
+
```

Comment:
Here are two templates (``ObjectTypeTemplate``, ``ObjectTypeArgsTemplate``) for objects, which we discussed to remake. The first template is needed as a part of pipe processing: the result of ObjectTypeArgsTemplate is passed to ObjectTypeTemplate and it makes the final composition. In case when ObjectTypeTemplate is null the result is not expectable.

Reply:
Left two comments above, but this thread can be resolved.